### PR TITLE
Flag stale PRs for engine to framework roller

### PR DESCRIPTION
**Pull request by keyonghan**
Following up of https://github.com/flutter/cocoon/pull/3332/files#r1423217355

This PR does:
1) supports engine to framework roller
2) change stale logic to focus on `queued` status. There are cases where GitHub check runs have finished a while ago, and we do not want to trigger alerts for them.

### CODE REVIEW

File: auto_submit/lib/validations/ci_successful.dart [hunk 1]
```diff
@@ -222,7 +222,19 @@ class CiSuccessful extends Validation {
     return dateTime.compareTo(DateTime.now().subtract(const Duration(hours: Config.kGitHubCheckStaleThreshold))) < 0;
   }
 
-  bool isEngineRoller(Author author, github.RepositorySlug slug) {
+  /// Perform stale check only on Engine related rolled PRs.
+  ///
+  /// This includes those rolled PRs from upstream to Engine repo and those
+  /// rolled PRs from Engine to Framework.
+  bool shouldStale(Author author, github.RepositorySlug slug) {
```
Comment (ricardoamador) [inline auto_submit/lib/validations/ci_successful.dart:229]: Should this be something like isStale rather than shouldStale?

Comment (ricardoamador) [reply auto_submit/lib/validations/ci_successful.dart:229]: or markStale?

Comment (keyonghan) [reply auto_submit/lib/validations/ci_successful.dart]: Updated to `supportStale`.

**Review by ricardoamador — APPROVED** [review]
LGTM

**Pull request merged**